packages/medusa/src/repositories/price-list.ts

```diff
@@ -1,39 +1,69 @@
-import { FindOptionsWhere, ILike } from "typeorm"
+import { FindOperator, FindOptionsWhere, ILike, In } from "typeorm"
 import { PriceList } from "../models"
-import { CustomFindOptions, ExtendedFindConfig } from "../types/common"
+import { ExtendedFindConfig } from "../types/common"
 import { dataSource } from "../loaders/database"
 
-export type PriceListFindOptions = CustomFindOptions<
-  PriceList,
-  "status" | "type"
->
-
 export const PriceListRepository = dataSource.getRepository(PriceList).extend({
   async listAndCount(
     query: ExtendedFindConfig<PriceList>,
     q?: string
   ): Promise<[PriceList[], number]> {
     const query_ = { ...query }
+    query_.relationLoadStrategy = "query"
+    query_.where = query.where as FindOptionsWhere<PriceList>
 
-    if (q) {
-      query_.where = query_.where as FindOptionsWhere<PriceList>
-      delete query_.where.description
-      delete query_.where.name
+    const groups = query_.where.customer_groups as unknown as FindOperator<
+      string[]
+    >
+    delete query_.where.customer_groups
+
+    if (groups) {
+      query_.relations = query_.relations ?? {}
+      query_.relations.customer_groups =
+        query_.relations.customer_groups ?? true
 
-      query_.where.description = ILike(`%${q}%`)
-      query_.where.name = ILike(`%${q}%`)
       query_.where.customer_groups = {
-        name: ILike(`%${q}%`),
+        ...(query_.where.customer_groups ?? {}),
+        id: In(groups.value),
+      }
+    }
+
+    if (q) {
+      query_.where = query_.where ?? {}
+      query_.where = [
+        {
+          ...query_.where,
+          name: ILike(`%${q}%`),
+        },
+        {
+          ...query_.where,
+          description: ILike(`%${q}%`),
+        },
+      ]
+
+      if (groups) {
+        query_.where.push({
+          ...query_.where,
+          customer_groups: {
+            id: In(groups.value),
+            name: ILike(`%${q}%`),
+          },
+        })
+      } else {
+        query_.relations = query_.relations ?? {}
+        query_.relations.customer_groups =
```

Comment:
q: why would customer_groups be true for all cases except search? We're doing this above in line 22 as well

Reply:
It is outdated ^^

Reply:
ahah, done done.

Reply:
i ve clean it up a bit top avoid redundancy and incoherences, wdyt?

Reply:
🤌